app/controllers/application_controller.rb

```diff
@@ -3,8 +3,24 @@
 class ApplicationController < Sinatra::Base
 
   configure do
+  	enable :sessions
+    set :session_secret, "password_security"
     set :public_folder, 'public'
     set :views, 'app/views'
   end
 
+  get '/' do
+  	erb :index
+  end
+
+  helpers do
+		def logged_in?
+			!!session[:id]
+		end
```

Comment:
It's better to rely on the successful call to `current_user` than only introspecting on the `session[:id]` value. In theory, a user might have an invalid `id` in their session which would make `logged_in?` return true even though that user would not be findable later. Also, because `current_user` is memoized, it makes sense to call it as early in the request cycle as possible, which generally is the first call `logged_in?`.

```ruby
def logged_in?
  !!current_user
end
```